src/native/managed/cdacreader/tests/MockMemorySpace.cs

```diff
@@ -106,24 +114,96 @@ public Builder AddHeapFragments(IEnumerable<HeapFragment> fragments)
             return this;
         }
 
-        public ReadContext Create()
+        private HeapFragment CreateContractDescriptor(int jsonLength, int pointerDataCount)
         {
-            if (_created)
-                throw new InvalidOperationException("Context already created");
-            GCHandle fragmentReaderHandle = default; ;
-            if (_heapFragments.Count > 0)
+            Span<byte> descriptor = stackalloc byte[_targetTestHelpers.ContractDescriptorSize];
```

Comment:
This seems odd. Why not just allocate the array and then assign it below. Both the stackalloc and the array seem unnecessary.

Reply:
yea I missed these while copy/pasting.  They're leftover from the earlier unsafe pointer based approach.

I checked, there aren't any more `stackalloc` in the tests